eng/containers/UbuntuNetCoreKeyring/Dockerfile

```diff
@@ -1,24 +1,53 @@
-FROM mcr.microsoft.com/dotnet/sdk:5.0-focal AS build
+FROM mcr.microsoft.com/powershell:7.1.4-ubuntu-focal AS build
 
 ENV \
     NO_AT_BRIDGE=1 \
     DOCKER_CONTAINER_NAME="ubuntu_netcore_keyring" \
-    PSModuleAnalysisCachePath=/var/cache/microsoft/powershell/PSModuleAnalysisCache/ModuleAnalysisCache \
-    POWERSHELL_DISTRIBUTION_CHANNEL=PSDocker-DotnetCoreSDK-Ubuntu-20.04
+     # Unset ASPNETCORE_URLS from aspnet base image
+    ASPNETCORE_URLS= \
+    # Do not generate certificate
+    DOTNET_GENERATE_ASPNET_CERTIFICATE=false \
+    # SDK version
+    DOTNET_SDK_VERSION=5.0.401 \
+    DOTNET_SDK_VERSION2=3.1.413 \
+    # Enable correct mode for dotnet watch (only mode supported in a container)
+    DOTNET_USE_POLLING_FILE_WATCHER=true \
+    # Skip extraction of XML docs - generally not useful within an image/container - helps performance
+    NUGET_XMLDOC_MODE=skip \
+    # PowerShell telemetry for docker image usage
+    POWERSHELL_DISTRIBUTION_CHANNEL=PSDocker-DotnetSDK-Ubuntu-20.04 \
+    # Setup Dotnet envs
+    DOTNET_ROOT=/usr/share/dotnet  \
+    PATH=$PATH:usr/share/dotnet    
 
 # Install GNOME keyring
 RUN apt-get update \
     && apt-get install -y \
         libsecret-1-dev \
         dbus-x11 \
         gnome-keyring \
-        python
+        python \
+        curl
 
-# Install PowerShell && .net core runtime 2.1
-RUN wget -q https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb \
-    && dpkg -i packages-microsoft-prod.deb \
-    && apt-get update \
-    && apt-get install -y powershell \
-    && pwsh --version \
-    && apt-get install -y dotnet-runtime-2.1 \
-    && dotnet --list-runtimes
+# Below taken from https://hub.docker.com/_/microsoft-dotnet-sdk
+# 5.0.401-focal-amd64, 5.0-focal-amd64, 5.0.401-focal, 5.0-focal	Dockerfile	Ubuntu 20.04	09/14/2021
+
+# Install .NET SDK
```

Comment:
Is there a reason you are installing .NET here as opposed to using a dotnet base image?

Reply:
Yes, I ran into some strange issues installing the latest version of PowerShell to the global tools - so this approach is the opposite. We start with the PowerShell core image and install .NET instead. Considering that the version of .NET is more important to us than the version of PowerShell, this approach also seems to give us more control to tweak .NET as needed.

Reply:
FYI you could also consider using a multi stage build, where you use both the powershell and .net base images, and copy the dotnet directories over into the destination image (see [docs](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds)). I think what you have here is fine though.